src/coreclr/jit/morph.cpp

```diff
@@ -10769,6 +10770,163 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
             return node;
         }
 
+#if defined(TARGET_XARCH)
+        case NI_AVX512F_Add:
+        case NI_AVX512BW_Add:
+        case NI_AVX512F_And:
+        case NI_AVX512DQ_And:
+        case NI_AVX512F_AndNot:
+        case NI_AVX512DQ_AndNot:
+        case NI_AVX512F_Or:
+        case NI_AVX512DQ_Or:
+        case NI_AVX512F_Xor:
+        case NI_AVX512DQ_Xor:
```

Comment:
This covers the core bitwise patterns around `kmask` registers. It notably doesn't factor in `knot`, `kshift`, or `kxnor` but those are somewhat less common and can be handled in the future.